docs/index.rst

```diff
@@ -158,6 +158,30 @@ pointing to where in the given object validation failed, and a human readable
      }
    }
 
+Read-only fields
+~~~~~~~~~~~~~~~~
+Many of the API's :ref:`data structures <data>` contain read-only fields provided in API responses, for example, API-generated ``id`` fields, or a resource's ``url`` field. If read only fields are provided by clients in API request bodies, the way these are handled depends on the request method.
+
+For ``PUT`` requests, in order to allow clients to more easily send back an updated description, clients providing read-only fields is allowed, but these fields will be ignored when replacing the resource.
```

Comment:
I would write `clients may include read-only fields` instead of `clients providing read-only fields is allowed`.